test/integration/http_server_options_test.lua

```diff
@@ -87,3 +87,80 @@ g.test_disable_keepalive_default = function(g)
     local httpd = g.httpd
     t.assert_equals(table.getn(httpd.options.disable_keepalive), 0)
 end
+
+g.before_test('test_idle_timeout_default', function()
+    g.httpd = helpers.cfgserv()
+    g.httpd:start()
+end)
+
+g.test_idle_timeout_default = function(g)
+    local httpd = g.httpd
+    t.assert_equals(httpd.options.idle_timeout, 0)
+end
+
+g.before_test('test_idle_timeout_is_set', function()
+    g.idle_timeout = 0.5
+    g.httpd = helpers.cfgserv({
+        idle_timeout = g.idle_timeout,
+    })
+    g.httpd:start()
+end)
+
+g.test_idle_timeout_is_set = function(g)
+    local conn_is_opened = false
+    local conn_is_closed = false
+    local httpd = g.httpd
+    g.httpd.internal.preprocess_client_handler = function() conn_is_opened = true end
+    g.httpd.internal.postprocess_client_handler = function() conn_is_closed = true end
+
+    t.assert_equals(httpd.options.idle_timeout, g.idle_timeout)
+
+    -- Set HTTP keepalive headers: Connection:Keep-Alive and
+    -- Keep-Alive:timeout=<keepalive_idle>. Otherwise HTTP client will send
+    -- "Connection:close".
+    local opts = {
+        keepalive_idle = 3600,
+        keepalive_interval = 3600,
+    }
+    local r = http_client.new():request('GET', helpers.base_uri .. '/test', nil, opts)
+    t.assert_equals(r.status, 200)
+    t.assert_equals(r.headers.connection, 'keep-alive')
+    t.assert_equals(conn_is_opened, true)
+
+    helpers.retrying({
+        timeout = g.idle_timeout * 2,
+    }, function()
+        if conn_is_closed ~= true then
+            assert()
+        end
+    end)
```

Comment:
It can be simpler:

```diff
diff --git a/test/integration/http_server_options_test.lua b/test/integration/http_server_options_test.lua
index 504891d..e5411a6 100644
--- a/test/integration/http_server_options_test.lua
+++ b/test/integration/http_server_options_test.lua
@@ -130,9 +130,7 @@ g.test_idle_timeout_is_set = function(g)
     helpers.retrying({
         timeout = g.idle_timeout * 2,
     }, function()
-        if conn_is_closed ~= true then
-            assert()
-        end
+        assert(conn_is_closed, 'assert that the connection is closed')
     end)
     t.assert_equals(conn_is_closed, true) -- Connection is closed.
 end
```